nil pointer error in getServiceURI() on startup

[Original report](https://github.com/livepeer/go-livepeer/issues/1780#issuecomment-788902131)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x10c48c5]

goroutine 1 [running]:
net/url.(*URL).Hostname(...)
        /usr/local/go/src/net/url/url.go:1112
main.getServiceURI(0xc0000dc160, 0x0, 0x0, 0x0, 0x0, 0x0)
        /build/cmd/livepeer/livepeer.go:1070 +0x2e5
main.main()
        /build/cmd/livepeer/livepeer.go:873 +0x5185
```

Reported on v0.5.14.

--- a/CHANGELOG_PENDING.md
+++ b/CHANGELOG_PENDING.md
@@ -26,5 +26,6 @@
 #### Orchestrator
 
 - \#1860 Discard low gas prices to prevent insufficient ticket faceValue errors (@kyriediculous)
+- \#1859 Handle error for invalid inferred orchestrator public IP on node startup (@reubenr0d)
 
 #### Transcoder

--- a/cmd/livepeer/livepeer.go
+++ b/cmd/livepeer/livepeer.go
@@ -1012,17 +1012,21 @@ func getServiceURI(n *core.LivepeerNode, serviceAddr string) (*url.URL, error) {
 	// TODO probably should put this (along w wizard GETs) into common code
 	resp, err := http.Get("https://api.ipify.org?format=text")
 	if err != nil {
-		glog.Error("Could not look up public IP address")
+		glog.Errorf("Could not look up public IP err=%v", err)
 		return nil, err
 	}
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		glog.Error("Could not look up public IP address")
+		glog.Errorf("Could not look up public IP err=%v", err)
 		return nil, err
 	}
 	addr := "https://" + strings.TrimSpace(string(body)) + ":" + RpcPort
 	inferredUri, err := url.ParseRequestURI(addr)
+	if err != nil {
+		glog.Errorf("Could not look up public IP err=%v", err)
+		return nil, err
+	}
 	if n.Eth == nil {
 		// we won't be looking up onchain sURI so use the inferred one
 		return inferredUri, err
@@ -1031,12 +1035,12 @@ func getServiceURI(n *core.LivepeerNode, serviceAddr string) (*url.URL, error) {
 	// On-chain lookup and matching with inferred public address
 	addr, err = n.Eth.GetServiceURI(n.Eth.Account().Address)
 	if err != nil {
-		glog.Error("Could not get service URI; orchestrator may be unreachable")
+		glog.Errorf("Could not get service URI; orchestrator may be unreachable err=%v", err)
 		return nil, err
 	}
 	ethUri, err := url.ParseRequestURI(addr)
 	if err != nil {
-		glog.Error("Could not parse service URI; orchestrator may be unreachable")
+		glog.Errorf("Could not parse service URI; orchestrator may be unreachable err=%v", err)
 		ethUri, _ = url.ParseRequestURI("http://127.0.0.1:" + RpcPort)
 	}
 	if ethUri.Hostname() != inferredUri.Hostname() || ethUri.Port() != inferredUri.Port() {

